lib/msf/ui/console/command_dispatcher/developer.rb

```diff
@@ -51,6 +63,19 @@ def reload_file(path)
     load path
   end
 
+  def reload_diff_files
+    unless git?
+      return
+    end
+    output, status = Open3.capture2e *%w(git diff --name-only)
+    if status.success?
+      files = output.split("\n")
+      files.each { |file| reload_file(file) }
```

Comment:
Does this list needs to filter modules?  If I remember correctly modules require a different reload path.

Reply:
Modules have a shitty regex to prevent them from being reloaded. It's handled in `reload_file`, too. I can probably do one better leveraging `Msf::Config`.

Reply:
Everything *should* work as is, however `print_error` will be called for each module in the list, so filtering before sending may avoid user confusion.

Reply:
https://github.com/rapid7/metasploit-framework/pull/10705#discussion_r220763405